OPHD/Things/Structures/RobotCommand.cpp

```diff
@@ -57,11 +57,5 @@ void RobotCommand::addRobot(Robot* robot)
  */
 void RobotCommand::removeRobot(Robot* robot)
 {
-	if (!robot) { throw std::runtime_error("RobotCommand::removeRobot() called with nullptr"); }
-	if (mRobotList.empty()) { return; }
-
-	auto iter = find(mRobotList.begin(), mRobotList.end(), robot);
-	if (iter == mRobotList.end()) { return; }
-
-	mRobotList.erase(iter);
+	mRobotList.erase(std::remove_if(mRobotList.begin(), mRobotList.end(), [robot](Robot* r1) { return r1 == robot; }), mRobotList.end());
```

Comment:
Oh yeah, I forgot the `std::erase`. I find that's an area where the STL is not super obvious.

----

If the lambda is just testing for equality, I believe the value overload could have been used instead.
```cpp
mRobotList.erase(std::remove_if(mRobotList.begin(), mRobotList.end(), robot), mRobotList.end());
```
See overload (1) instead of overload (3): [`std::remove`](https://en.cppreference.com/w/cpp/algorithm/remove)

Just noticing now I should have said `std::remove` earlier instead of `std::remove_if`.

Reply:
Yeah, I thought that `std::remove_if` was overkill? But I wanted to practice my lambda expressions since I'm still a little unfamiliar with them. Going to commit the change with `std::remove` and call this one done. :D